numpyro/distributions/iaf.py

```diff
@@ -0,0 +1,84 @@
+# adapted from https://github.com/pyro-ppl/pyro/blob/dev/pyro/distributions/transforms/iaf.py
+
+from __future__ import absolute_import, division, print_function
+
+from numpyro.distributions.constraints import Transform, real
+from jax.lax import stop_gradient
+import jax.numpy as np
+
+
+def clamp_preserve_gradients(x, min, max):
```

Comment:
Could you rename this to `"_clamp..."` so as to signify private usage? Otherwise, we can add it to `distributions.util` in case this will be more generally useful.

Reply:
ok